blocks/basic/test/qa_sources.cpp

```diff
@@ -146,6 +162,68 @@ const boost::ut::suite TagTests = [] {
         sched.runAndWait();
         expect(eq(n_samples, static_cast<std::uint32_t>(sink.n_samples_produced))) << "Number of samples does not match";
     };
+
+    "FunctionGenerator ImChart test"_test = [] {
+        const std::size_t        N           = 128; // test points
+        double                   start_value = 10.;
+        double                   final_value = 20.;
+        std::vector<std::string> signals{ "Const", "LinearRamp", "ParabolicRamp", "CubicSpline", "ImpulseResponse" };
+        for (const auto &sig : signals) {
+            FunctionGenerator<double> funcGen{};
+            std::ignore = funcGen.settings().set({ { "signal_type", sig },
```

Comment:
Please don't 'std::ignore' keep the return and assert that it's `empty()` meaning that all settings have been accepted by the block (nice regression unit-test case)